### PR TITLE
Fix failed ios build after second info-plist merge

### PR DESCRIPTION
**Pull request by Fatme**
Fixes https://github.com/NativeScript/nativescript-cli/issues/1066

### CODE REVIEW

File: lib/services/android-project-service.ts [hunk 1]
```diff
@@ -146,6 +146,17 @@ class AndroidProjectService extends projectServiceBaseLib.PlatformProjectService
...
 			let gradleSettingsFilePath = path.join(this.platformData.projectRoot, "settings.gradle");
 			shell.sed('-i', /__PROJECT_NAME__/, this.getProjectNameFromId(), gradleSettingsFilePath);
+		}).future<void>()();
+	}
+
+	public interpolateConfigurationFile(): IFuture<void> {
+		return (() => {
+			let manifestPath = this.platformData.configurationFilePath;
+
+			console.log("ANDROID MANIFEST FILE PATH!!!!");
```
Comment (rosen-vladimirov) [inline lib/services/android-project-service.ts]: do you need these console.logs

File: lib/services/plugins-service.ts [hunk 2]
```diff
@@ -368,6 +351,27 @@ export class PluginsService implements IPluginsService {
 		}).future<void>()();
 	}
 
+	private merge(pluginData: IPluginData, platformData: IPlatformData): IFuture<void> {
+		return (() => {
+			let tnsModulesDestinationPath = path.join(platformData.appDestinationDirectoryPath, constants.APP_FOLDER_NAME, constants.TNS_MODULES_FOLDER_NAME);
+			let nodeModules = this.$broccoliBuilder.getChangedNodeModules(tnsModulesDestinationPath, platformData.normalizedPlatformName.toLowerCase()).wait();
+
+			_(_.keys(nodeModules))
```
Comment (rosen-vladimirov) [inline lib/services/plugins-service.ts]: why not
```
_(nodeModules)
     .keys()
     ....
```

File: lib/services/ios-project-service.ts [hunk 3]
```diff
@@ -118,20 +119,26 @@ export class IOSProjectService extends projectServiceBaseLib.PlatformProjectServ
...
+			this.replaceFileName("-Prefix.pch", path.join(this.platformData.projectRoot, IOSProjectService.IOS_PROJECT_NAME_PLACEHOLDER)).wait();
+			this.replaceFileName(IOSProjectService.XCODE_PROJECT_EXT_NAME, this.platformData.projectRoot).wait();
 
-			let pbxprojFilePath = path.join(projectRoot, this.$projectData.projectName + IOSProjectService.XCODE_PROJECT_EXT_NAME, "project.pbxproj");
+			let pbxprojFilePath = path.join(this.platformData.projectRoot, this.$projectData.projectName + IOSProjectService.XCODE_PROJECT_EXT_NAME, "project.pbxproj");
 			this.replaceFileContent(pbxprojFilePath).wait();
 		}).future<void>()();
 	}
 
+	public interpolateConfigurationFile(configurationFilePath?: string): IFuture<void> {
```
Comment (teobugslayer) [inline lib/services/ios-project-service.ts]: Is there a reason for this function to return a future?

**Comment by teobugslayer** [comment]
Why there are changes in the Android specific code? Can you document them in the PR description, or separate them in another PR?

File: lib/services/ios-project-service.ts [hunk 4]
```diff
@@ -118,20 +119,26 @@ export class IOSProjectService extends projectServiceBaseLib.PlatformProjectServ
...
 		return (() => {
-			let infoPlistFilePath = path.join(projectRoot, IOSProjectService.IOS_PROJECT_NAME_PLACEHOLDER, util.format("%s-%s", IOSProjectService.IOS_PROJECT_NAME_PLACEHOLDER, "Info.plist"));
-			shell.sed('-i', "__CFBUNDLEIDENTIFIER__", this.$projectData.projectId, infoPlistFilePath);
+			let infoPlistFilePath = path.join(this.platformData.projectRoot, IOSProjectService.IOS_PROJECT_NAME_PLACEHOLDER, util.format("%s-%s", IOSProjectService.IOS_PROJECT_NAME_PLACEHOLDER, "Info.plist"));
+			this.interpolateConfigurationFile(infoPlistFilePath).wait();
 
-			this.replaceFileName("-Info.plist", path.join(projectRoot, IOSProjectService.IOS_PROJECT_NAME_PLACEHOLDER)).wait();
-			this.replaceFileName("-Prefix.pch", path.join(projectRoot, IOSProjectService.IOS_PROJECT_NAME_PLACEHOLDER)).wait();
-			this.replaceFileName(IOSProjectService.XCODE_PROJECT_EXT_NAME, projectRoot).wait();
+			this.replaceFileName("-Info.plist", path.join(this.platformData.projectRoot, IOSProjectService.IOS_PROJECT_NAME_PLACEHOLDER)).wait();
```
Comment (teobugslayer) [inline lib/services/ios-project-service.ts]: You can extract this long path into a dedicated variable - it is used on more than one place

File: lib/services/ios-project-service.ts (hunk 3 above)
Comment (Fatme) [inline lib/services/ios-project-service.ts]: Yes - https://github.com/NativeScript/nativescript-cli/pull/1083/files#diff-b64860922c31aeed973aef6663b2a536R152

**Comment by Fatme** [comment]
This is the reason why I need the changes in the Android specific code - https://github.com/NativeScript/nativescript-cli/pull/1083/files#diff-513fe963f5d821a1cdb167ea91d70f1cR105. I cannot separate this in another PR, this change must be here.

**Comment by dtopuzov** [comment]
:+1:

**Pull request merged**